check-windows-eventlog/check-windows-eventlog.go

```diff
@@ -293,6 +371,30 @@ func (opts *logOpts) searchLog(logName string) (warnNum, critNum int64, errLines
 		}
 		lastNumber = r.RecordNumber
 
+		if opts.idRangeList != nil {
+			// even code takes last 4 bytes
+			eventID := r.EventID & 0x0000FFFF
+			found := false
+			exact := false
+			for _, idr := range opts.idRangeList {
+				if !idr.bang {
+					exact = true
+					if idr.lo <= eventID && eventID <= idr.hi {
+						found = true
+					}
+				} else {
+					if idr.lo <= eventID && eventID <= idr.hi {
+						found = false
+					} else if !exact {
+						found = true
```

Comment:
thanks for your review.
hmm, this code is confusible. :(

|No|Seq  |Exact|Found|
|--|-----|-----|-----|
| 1|!2   |false|false|
| 2|!2,!3|false|found|

So `exact` should be true when the first `!` found.

```go
if !idr.bang {
	exact = true
	if idr.lo <= eventID && eventID <= idr.hi {
		found = true
	}
} else {
	if idr.lo <= eventID && eventID <= idr.hi {
		found = false
		exact = true
	} else if !exact {
		found = true
	}
}
```

I'll add test for this edge case. Thanks.